website/src/components/SearchPage/Table.tsx

```diff
@@ -136,6 +157,20 @@ export const Table: FC<TableProps> = ({
                                 onClick={(e) => handleRowClick(e, row[primaryKey] as string)}
                                 onAuxClick={(e) => handleRowClick(e, row[primaryKey] as string)}
                             >
+                                <td
+                                    className='px-2 whitespace-nowrap text-primary-900 md:pl-6'
+                                    onClick={(e) => {
+                                        e.stopPropagation();
+                                        handleRowSelectToggle(e, row[primaryKey] as string);
```

Comment:
I think rather than just toggling its nice to really link with the state of the HTML element: https://chatgpt.com/share/66fc61c9-5fa4-8005-8e16-a492aaa0f10f

Reply:
(that may or may not solve the issue Chaoran and I are seeing)

Reply:
I'm not quite sure what you mean I think. The checkbox `checked` property is linked up with the state.

I trie to be clever and `preventDefault` on the click event in the checkbox and then bubble it up, but that prevented the checkbox from showing checked. Now I changed it to simply call teh handleRowSelectToggle from both elements (the checkbox and the cell). If you have a good idea how to have the call only in a single handler, let me know!

Reply:
https://github.com/loculus-project/loculus/pull/2917/commits/9cbf3a2e8867d9e469f25a4644f82b6595c195ed

Reply:
Sorry, yes I should have explained better. What I meant was making use of `e.target.checked` is the handler so that when setting you are actually checking the HTML, rather than assuming the HTML matches the React state. I'm not particularly doubting that it will match the React state but it just feels like good practice.

Reply:
Ah, I think I understand now. Have a look again?

https://github.com/loculus-project/loculus/pull/2917/commits/4c67c812c29bd1dd04724d3cece578713ae1aec3

I'm still not happy with calling the handler twice (once from the `td` onClick and once from the checkbox onChange) but I think there isn't a clean way around it.

Reply:
Ah I think I hadn't understood the issue. I'm afk ATM but it sounds like the solution could be stopPropagation instead of prevent default. I'll check back later

Reply:
The issue I have is that I want the cell to also be clickable, and a cell click should also check the checkbox. But if I call my handler function from the onClick of the cell, as well as from the onChange of the checkbox, if at some point the handlerfunction gets change it's easy to overlook that it is called in two places.

I'd prefer if I could just call it from the checkbox for example, and from the cell simply trigger the change on the checkbox and then have the onChange handler on the checkbox do the rest. But I tried that by manually dispatching a change event to the checkbox, but that doesn't work because the onChange react handler doesn't get triggered by that.

Another way could be to only to the handling in the cell, and from the checkbox bubble up something instead, but that also feels messy. So for now I settled on having the function call duplicated -- it's only 5 lines apart so it should be fine I think.

Reply:
Ah I hadn't understood this was about making the cell clickable. I guess that does seem nice for UX - cool!